kotlin/add-graphics-renderer/src/main/java/com/esri/arcgisruntime/sample/addgraphicsrenderer/MainActivity.kt

```diff
@@ -0,0 +1,131 @@
+/* Copyright 2020 Esri
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package com.esri.arcgisruntime.sample.addgraphicsrenderer
+
+import android.graphics.Color
+import android.os.Bundle
+import androidx.appcompat.app.AppCompatActivity
+import com.esri.arcgisruntime.geometry.Point
+import com.esri.arcgisruntime.geometry.PolygonBuilder
+import com.esri.arcgisruntime.geometry.PolylineBuilder
+import com.esri.arcgisruntime.geometry.SpatialReferences
+import com.esri.arcgisruntime.mapping.ArcGISMap
+import com.esri.arcgisruntime.mapping.Basemap
+import com.esri.arcgisruntime.mapping.view.Graphic
+import com.esri.arcgisruntime.mapping.view.GraphicsOverlay
+import com.esri.arcgisruntime.symbology.SimpleFillSymbol
+import com.esri.arcgisruntime.symbology.SimpleLineSymbol
+import com.esri.arcgisruntime.symbology.SimpleMarkerSymbol
+import com.esri.arcgisruntime.symbology.SimpleRenderer
+import kotlinx.android.synthetic.main.activity_main.*
+
+class MainActivity : AppCompatActivity() {
+
+  override fun onCreate(savedInstanceState: Bundle?) {
+    super.onCreate(savedInstanceState)
+    setContentView(R.layout.activity_main)
+
+    // create a map with the Basemap Type topographic
+    val map = ArcGISMap(Basemap.Type.TOPOGRAPHIC, 15.169193, 16.333479, 2)
+    // add graphics overlay
+    addGraphicsOverlay()
+    // set the map to be displayed in this view
+    mapView.map = map
+  }
+
+  private fun addGraphicsOverlay() {
+    // point graphic
+    // create a graphic overlay for the point
+    val pointGraphicOverlay = GraphicsOverlay()
```

Comment:
I think we can neaten this up a little, and the same for the other graphics overlay set up.  something like

```
    // create a graphic overlay for the point
    val pointGraphicOverlay = GraphicsOverlay().apply {
      // add the point graphic to the graphics overlay
      graphics.add(pointGraphic)
      // set the renderer on the graphics overlay to the new renderer
      renderer = pointRenderer
    }
```

Reply:
How do you feel about keeping the `mapView.graphicsOverlays.add(...)` separate for each graphic and using this pattern?

```
// create graphic overlay for polygon
GraphicsOverlay().let {
      // add graphic to overlay
      it.graphics.add(polygonGraphic)
      // set the renderer on the graphics overlay to the new renderer
      it.renderer = polygonRenderer
      // add graphics overlay to MapView
      mapView.graphicsOverlays.add(it)
    }
```

Combined with my other suggestion above ie. make each graphic its own function, it might end up something like

```
private fun createPointGraphicsOverlay(): GraphicsOverlay = 
GraphicsOverlay().let {
      // create graphic for polygon with a geometry
      val polygonGraphic =   
        Graphic(PolygonBuilder(SpatialReferences.getWebMercator()).apply {
            addPoint(-20e5, 20e5)
            addPoint(20e5, 20e5)
            addPoint(20e5, -20e5)
            addPoint(-20e5, -20e5)
        }.toGeometry())

      // add graphic to overlay
      it.graphics.add(polygonGraphic)

      // set the renderer on the graphics overlay to a new renderer with a symbol
      it.renderer = SimpleRenderer(SimpleFillSymbol(SimpleFillSymbol.Style.SOLID, Color.YELLOW, null))

      // add graphics overlay to MapView
      mapView.graphicsOverlays.add(it)
    }
```

Reply:
yes good thinking! Can you push up those suggestions and I can take a look? Only suggestion would be not to include the `val polygonGraphic` set up within the `.let` block, that doesn't need to be within the scoping function (which should be kept as clean and relevant as possible). It could be moved to just about the `let` call. Also if you could break the steps up for the polygonGraphic as in your original PR: it's a bit more readable than bunching it all together.